src/ucp/tag/rndv.c

```diff
@@ -18,6 +18,27 @@ static int ucp_rndv_is_get_zcopy(ucp_request_t *sreq, ucp_rndv_mode_t rndv_mode)
             ((rndv_mode == UCP_RNDV_MODE_AUTO) &&
              UCP_MEM_IS_HOST(sreq->send.mem_type)));
 }
+static int ucp_rndv_is_pipeline_needed(ucp_request_t *sreq) {
+    ucp_rsc_index_t rsc_index;
+    ucp_rndv_mode_t rndv_mode;
+
+    rndv_mode = sreq->send.ep->worker->context->config.ext.rndv_mode;
+    rsc_index = ucp_ep_config(sreq->send.ep)->key.lanes[sreq->send.lane].rsc_index;
+
+    if ((rndv_mode == UCP_RNDV_MODE_PUT_ZCOPY) ||
+        ((rndv_mode == UCP_RNDV_MODE_AUTO) &&
+         UCP_MEM_IS_HOST(sreq->send.mem_type))) {
+        return 0;
+    }
+
+    if (strstr(sreq->send.ep->worker->context->tl_rscs[rsc_index].
```

Comment:
need to detect this independent of the transport name, for example check if the transport supports the memory type of the buffer

Reply:
I think checking if transport supports memory type is not enough here.  for example IB TLs supports  CUDA mem type  but need pipe lining.  whereas, cuda-ipc  don't need pipelining.

yeah. its bit hacky, is there any better way?

Reply:
is there some way to tell in UCT API that cuda-ipc does not need pipelining? is it because of pci p2p performance? maybe we can use iface_attr->bw to select the one with best score?

Reply:
yes, because pci p2p performance with GPUDirectRDMA.  I think we cannot use  iface_attr->bw here because both these lanes (cuda_ipc and IB) are not in same EP.

On the same RNDV protocol,
we need pipelining on Inter-Node  GPU-GPU transfer over IB.  
we don't need pipelining when Intra-Node  GPU-GPU over cuda-IPC

We have updates/enhancements in the pipelining protocol to handle the more GPU/HCA configurations for Oct release.  MAy be I will revisit this item and see if we can tell in UCT API if pipelining is needed or not.

Reply:
so it means that when cuda-ipc is available it's always preferred?
why not set the BW of cuda-ipc to higher than IB, and then select pipelined proto if the min BW of its components is higher than the alternative?

Reply:
not sure If I'm missing something.  cuda-ipc and  IB are not alternative. they won't be there in same EP.  cuda-ipc is intra-node and  IB is inter-node.